iOS/src/toga_iOS/hardware/location.py

```diff
@@ -139,8 +168,36 @@ def start_tracking(self):
         self.native.pausesLocationUpdatesAutomatically = False
 
         self._is_tracking = True
+        self.tracking_mode = LocationMode.CONTINUOUS
         self.native.startUpdatingLocation()
 
+    def start_significant_tracking(self) -> None:
+        """Start monitoring significant location changes."""
+        # Ensure that background processing will occur
+        self.native.allowsBackgroundLocationUpdates = True
+        self.native.pausesLocationUpdatesAutomatically = False
+
+        self._is_tracking = True
+        self.tracking_mode = LocationMode.SIGNIFICANT
+        self.native.startMonitoringSignificantLocationChanges()
+
+    def start_visit_tracking(self) -> None:
+        """Start monitoring visits (CLVisit events)."""
+        # Ensure that background processing will occur
+        self.native.allowsBackgroundLocationUpdates = True
+        self.native.pausesLocationUpdatesAutomatically = False
+
+        self._is_tracking = True
+        self.tracking_mode = LocationMode.VISITS
+        self.native.startMonitoringVisits()
+
     def stop_tracking(self):
-        self.native.stopUpdatingLocation()
         self._is_tracking = False
+        if self.tracking_mode == LocationMode.CONTINUOUS:
+            self.native.stopUpdatingLocation()
+        elif self.tracking_mode == LocationMode.SIGNIFICANT:
+            self.native.stopMonitoringSignificantLocationChanges()
+        elif self.tracking_mode == LocationMode.VISITS:
+            self.native.stopMonitoringVisits()
+        else:
+            return
```

Comment:
As a stylistic thing - if we know tracking mode can only be one of the enum values there's no need for the `else` case here; it will be reported as a coverage failure. Instead, make the "default" value the bucket `else` case.